lib/test_lib.py

```diff
@@ -571,7 +571,7 @@ def SetupClients(self, nr_clients):
         fd.Flush()
 
         index.AddClient(fd)
-
+    index.Close()
```

Comment:
No, the ClientIndex aff4 object writes all data directly to the DB unless you explicitly give it sync=False. The problem you encountered was not that the data wasn't there but that when you called Open on the MAIN_INDEX urn, there was no aff4 type there so you got back an AFF4Volume instead of an index and that would not know about all the functions you called afterwards. Just give it a try and see if it works?